### ISSUE TITLE
Batch Training and DataLoader in PyG

### ISSUE THREAD

**Issue by khaled-rahman** [pyg-team/pytorch_geometric#3155]
### Discussed in https://github.com/pyg-team/pytorch_geometric/discussions/3154

<div type='discussions-op-text'>

<sup>Originally posted by **khaled-rahman** September 16, 2021</sup>
Hi, I was trying to train a model in PyTorch Lightning for large graphs.
To create a batch, I used `NeighborSampler` following pl [examples](https://github.com/pyg-team/pytorch_geometric/blob/ffaf0922d0d1cdb85a89084d759e61ae5735250e/examples/pytorch_lightning/graph_sage.py).
To create the batch graph, is it possible to take all neighbors instead of size parameters?
More specifically, I would like to create the batch on nodes, e.g., if node_ids = [0, 1, 2, 3, 4, ..., 49], and batch_size=4, then I would like to take node ids [0, 1, 2, 3] in a batch, [4, 5, 6, 7] in the second batch and so on. Also, the edge_index for the first batch should contain all the edges of a graph that are incoming/outgoing from the nodes [0, 1, 2, 3], i.e., `Data(x = [0, 1, 2, 3], edge_index = [[0, 1, 2, 2, 3,. ...], [2, 3, 5, 6, 0, ...]])`

Is it possible to create such batches using `NeighborSample` or `NeighborLoader`?
It would be helpful to get some pointers on this.</div>

**Comment by rusty1s** [comment]
If you want to create mini-batches in order, you can set `shuffle=False`. If you want to also do no sampling at all, i.e., sample the full subgraph, you can set `sizes`/`num_neighbors` to `-1`, e.g., `[-1, -1]` will sample the full two-hop neighborhood around each node in the mini-batch. Keep in mind that this can result in really big subgraphs for nodes with high node degree :)

**Comment by khaled-rahman** [comment]
Thank you so much! This is really helpful to know.
